File: trunk/src/java/org/codehaus/testdox/intellij/actions/BaseTestElementAction.java
package org.codehaus.testdox.intellij.actions;

import javax.swing.Icon;

import com.intellij.openapi.actionSystem.AnActionEvent;

abstract class BaseTestElementAction extends BaseAction {

    protected final boolean useFromTestDoxToolWindow;

    BaseTestElementAction(String text, String description, Icon icon, boolean useFromTestDoxToolWindow) {
        super(text, description, icon);
        this.useFromTestDoxToolWindow = useFromTestDoxToolWindow;
    }

    public void update(AnActionEvent event) {
        if (useFromTestDoxToolWindow) {
            actionEvents().getTestDoxToolWindowUI(event).updatePresentation(event.getPresentation());
        } else {
            actionEvents().getTestDoxController(event).updatePresentation(event.getPresentation());
        }
    }
}

File: trunk/src/scala/org/codehaus/testdox/intellij/actions/BaseTestElementAction.scala
package org.codehaus.testdox.intellij.actions

import javax.swing.Icon

import com.intellij.openapi.actionSystem.AnActionEvent

abstract class BaseTestElementAction(text: String, description: String, icon: Icon, protected val useFromTestDoxToolWindow: Boolean)
    extends BaseAction(text, description, icon) {

  override def update(event: AnActionEvent) {
    selectController(event).update(event.getPresentation)
  }

  private def selectController(event: AnActionEvent) = {
    if (useFromTestDoxToolWindow)
      actionEvents.getToolWindowUI(event)
    else
      actionEvents.getTestDoxController(event)
  }
}
